doc/api/crypto.md

```diff
@@ -2651,6 +2651,18 @@ added: v10.0.0
 Enables the FIPS compliant crypto provider in a FIPS-enabled Node.js build.
 Throws an error if FIPS mode is not available.
 
+### crypto.signOneShot(data, privateKey)
+<!-- YAML
+added: REPLACEME
+-->
+* `data` {Buffer | TypedArray | DataView}
+* `privateKey` {Buffer | KeyObject}
```

Comment:
This deviates from other functions that consume keys, where they can also be `string`s and certain `object`s.

Reply:
Supporting string encoding names complicates things and makes separating required vs. optional function arguments nearly impossible as far as I can tell.

Besides, the only reason we have the string encoding parameters is for backwards compatibility and internally node immediately converts it to a `Buffer` using `Buffer.from()`, something userland could easily do themselves these days...

Reply:
I'm not talking about the `data`, only about `privateKey`. See e.g. `privateDecrypt`:

> If privateKey is not a KeyObject, this function behaves as if privateKey had been passed to crypto.createPrivateKey().

I designed that and I still think it is a good idea while maintaining compatibility with older API designs.

Reply:
To me it should be all or nothing, it doesn't make sense to have an encoding parameter for just one parameter when we're discussing backwards compatibility (`.update()` also supports an encoding parameter).

Reply:
Again, I am not talking about the `data` parameter and none of the existing APIs accept an `encoding` option or argument for parsing the key, since PEM is ASCII-only.

As far as I can tell, your code already accepts strings and objects as the `privateKey`, the documentation does not match that behavior, that's all I am trying to explain.

Reply:
Doc updated.